integrade/tests/api/v1/test_user_accounts.py

```diff
@@ -129,3 +129,22 @@ def test_change_password():
 
     response = client.post(urls.AUTH_TOKEN_DESTROY, {}, auth=auth)
     assert response.status_code == 204
+
+
+def test_cannot_delete_self():
+    """Ensure an user cannot delete their own account."""
+    user = create_user_account()
+    payload = {
+        'current_password': user['password'],
+    }
+
+    auth = get_auth(user)
+    client = api.Client(authenticate=False)
+
+    response = client.get(urls.AUTH_ME, auth=auth)
+    assert response.status_code == 200, 'User as not setup properly'
+
+    client.post(urls.AUTH_USERS_DELETE, payload, auth=auth)
+
+    response = client.get(urls.AUTH_ME, auth=auth)
+    assert response.status_code == 200, 'User was allowed to delete themselves'
```

Comment:
Just reading this is confusing me, because it looks like if the response to `AUTH_ME` was 200, the user still existed....so it did not get deleted. I'll try it out myself I suppose.